python/lsst/ts/standardscripts/auxtel/calibrations/power_on_atcalsys.py

```diff
@@ -0,0 +1,289 @@
+                domain=self.domain,
+                name="ATMonochromator",
+                log=self.log,
+            )
+
+        await asyncio.gather(
+            self.white_light_source.start_task,
+            self.monochromator.start_task,
+        )
+
+        self.log.info("Configure completed")
+
+    def set_metadata(self, metadata):
+        """Compute estimated duration.
+
+        Parameters
+        ----------
+        metadata : SAPY_Script.Script_logevent_metadataC
+        """
+        metadata.duration = self.timeout_chiller_cool_down + self.timeout_lamp_warm_up
+
+    async def run(self):
+        """Run script."""
+        await self.assert_components_enabled()
+
+        await self.checkpoint("Starting chiller")
+        await self.start_chiller()
+
+        await self.checkpoint("Waiting for chiller to cool to set temperature")
+        await self.wait_for_chiller_temp_within_tolerance()
+
+        await self.checkpoint("Opening the shutter")
+        await self.open_white_light_shutter()
+
+        await self.checkpoint("Turning on lamp")
+        await self.switch_lamp_on()
+
+        await self.checkpoint("Waiting for lamp to warm up")
+        await self.wait_for_lamp_to_warm_up()
+
+        await self.checkpoint("Configuring ATMonochromator")
+        await self.configure_atmonochromator()
+
+    async def start_chiller(self):
+        await self.white_light_source.cmd_setChillerTemperature.set_start(
+            temperature=self.chiller_temperature, timeout=self.cmd_timeout
+        )
+        await self.white_light_source.cmd_startChiller.set(
+            timeout=self.timeout_chiller_cool_down
+        )
+
+    async def wait_for_chiller_temp_within_tolerance(self):
+        start_chill_time = time.time()
+        while time.time() - start_chill_time < self.timeout_chiller_cool_down:
+            chiller_temps = await self.white_light_source.tel_chillerTemperatures.aget()
+            tel_chiller_temp = chiller_temps.supplyTemperature
+            if (
+                abs(chiller_temps.setTemperature - tel_chiller_temp)
+                / chiller_temps.setTemperature
+                <= self.chiller_temp_tolerance_relative
+            ):
+                chill_time = time.time() - start_chill_time
+                self.log.info(
+                    f"Chiller reached target temperature {tel_chiller_temp}"
+                    f"within tolerance in {chill_time} s"
+                )
+                break
+            else:
+                continue
+        else:
+            raise TimeoutError(
+                f"Gave up waiting after {self.timeout_chiller_cool_down}"
+                f"for the chiller to chill to {self.chiller_temperature}."
+                f"Stayed at {tel_chiller_temp}"
+            )
+
+    async def open_white_light_shutter(self):
+        await self.white_light_source.cmd_openShutter.set_start()
+        # Q? Should we include a check that the shutter is open?
```

Comment:
I think this command only finishes when the shutter is open, and fails if the shutter doesn't open.. So  think it is ok to leave it as is, e.g., no need to check.